src/arr/trove/statistics.arr

```diff
@@ -222,9 +224,15 @@ end
 
 # please see: https://online.stat.psu.edu/stat462/
 
-fun multiple-regression(x_s_s :: List<Any>, y_s :: List<Number>) -> (Any -> Number):
+fun multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Number>) -> (Any -> Number):
   doc: "multiple-regression"
-  MR.multiple-regression(x_s_s, y_s)
+  mx_X = MX.lists-to-matrix(x_s_s.map(lam(x_s): x_s.push(1) end))
```

Comment:
This would be better written, I think, as 
```
intercepts = MX.col-matrix.make(raw-array-of(1, y_s.length())
mx_X = MX.lists-to-matrix(x_s_s).augment(intercepts)
```
instead of making tons of `link`ed lists, it'll directly make the raw arrays, and also make the mathematical structure more explicit.

Reply:
The `1`s should actually be in a new first column, so I tried inverting the arguments to the `augment`, but it doesn't seem to do the expected thing. Would you expect the following check to be "shipshape"? It doesn't for me. 

To check: add this to the end of `test-matrices.arr`, and do a `make test`.

I do note that `test-matrices.arr` already has some tests for `augment`. They seem to work. I don't know why this one fails.  Specifically, it throws away the last column and duplicates the penultimate column. Hope that immediately suggests what the bug is.

```
x_s_s = [list: [list: 4, 3], [list: 9/2, 2], [list: 5, 6/5], [list: 11/2, 9/2], [list: 6, 33/10], [list: 13/2, 10], [list: 7, 0]]
y_s = [list: 33, 42, 45, 51, 53, 61, 62]


intercepts = M.col-matrix.make(raw-array-of(1, y_s.length()))
mx_X0 = M.lists-to-matrix(x_s_s)
mx_X = M.mtx-augment(intercepts, mx_X0)

check "stats":
  mx_X is [matrix(7,3): 1, 4, 3,
                        1, 9/2, 2, 
                        1, 5, 6/5,
                        1, 11/2, 9/2,
                        1, 6, 33/10,
                        1, 13/2, 10,
                        1, 7, 0]
end
```

Reply:
Found it; fix incoming.

Reply:
See https://github.com/brownplt/pyret-lang/commit/eeef8a7a7235f688558bc3d0eeb3fe0c95c8f5fd

Reply:
Thanks! I've updated my PR accordingly.